scipy/_lib/_testutils.py

```diff
@@ -195,12 +197,17 @@ def test_views(self):
             args_array.append(self.arguments[arg_idx][0][::-1][::-1])
         self.pythranfunc(*args_array)
 
-    def test_strided(self):
-        args_array = []
-        for arg_idx in self.arguments:
-            args_array.append(np.repeat(self.arguments[arg_idx][0],
-                                        2, axis=0)[::2])
-        self.pythranfunc(*args_array)
+    @pytest.fixture
```

Comment:
This is a problem - we can't do `import pytest` at the top of this file nor directly under `_TestPythranFunc`, because `pytest` isn't a runtime dependency. 

`_TestPythranFunc` is only used to test a single function (`stats.somersd`), so apparently there's a regular thread safety issue there.

Reply:
Okay I found the problem, it's crashing:
```
$ pixi r test-nogil -t scipy.stats.tests.test_hypotests::TestSomersD -- --parallel-threads=40 -v
scipy/stats/tests/test_hypotests.py::TestSomersD::test_strided Fatal Python error: Aborted

Current thread 0x00007058d8ff96c0 (most recent call first):
  File "/home/rgommers/code/pixi-dev-scipystack/scipy/.pixi/envs/free-threading/lib/python3.13t/site-packages/numpy/_core/fromnumeric.py", line 57 in _wrapfunc
  File "/home/rgommers/code/pixi-dev-scipystack/scipy/.pixi/envs/free-threading/lib/python3.13t/site-packages/numpy/_core/fromnumeric.py", line 506 in repeat
  File "/home/rgommers/code/pixi-dev-scipystack/scipy/scipy/build-nogil-install/lib/python3.13t/site-packages/scipy/_lib/_testutils.py", line 201 in test_strided
  File "/home/rgommers/code/tmp/pytest-run-parallel/src/pytest_run_parallel/plugin.py", line 59 in closure
  File "/home/rgommers/code/pixi-dev-scipystack/scipy/.pixi/envs/free-threading/lib/python3.13t/threading.py", line 992 in run
  File "/home/rgommers/code/pixi-dev-scipystack/scipy/.pixi/envs/free-threading/lib/python3.13t/threading.py", line 1041 in _bootstrap_inner
  File "/home/rgommers/code/pixi-dev-scipystack/scipy/.pixi/envs/free-threading/lib/python3.13t/threading.py", line 1012 in _bootstrap

Thread 0x00007058d97fa6c0 (most recent call first):
  File "/home/rgommers/code/pixi-dev-scipystack/scipy/scipy/build-nogil-install/lib/python3.13t/site-packages/scipy/stats/_hypotests.py", line 689 in _somers_d
  File "/home/rgommers/code/pixi-dev-scipystack/scipy/scipy/build-nogil-install/lib/python3.13t/site-packages/scipy/stats/_hypotests.py", line 886 in somersd
  File "/home/rgommers/code/pixi-dev-scipystack/scipy/scipy/build-nogil-install/lib/python3.13t/site-packages/scipy/stats/tests/test_hypotests.py", line 682 in pythranfunc
  File "/home/rgommers/code/pixi-dev-scipystack/scipy/scipy/build-nogil-install/lib/python3.13t/site-packages/scipy/_lib/_testutils.py", line 203 in test_strided
  File "/home/rgommers/code/tmp/pytest-run-parallel/src/pytest_run_parallel/plugin.py", line 59 in closure
  File "/home/rgommers/code/pixi-dev-scipystack/scipy/.pixi/envs/free-threading/lib/python3.13t/threading.py", line 992 in run
  File "/home/rgommers/code/pixi-dev-scipystack/scipy/.pixi/envs/free-threading/lib/python3.13t/threading.py", line 1041 in _bootstrap_inner
  File "/home/rgommers/code/pixi-dev-scipystack/scipy/.pixi/envs/free-threading/lib/python3.13t/threading.py", line 1012 in _bootstrap

Thread 0x00007058d9ffb6c0 (most recent call first):
  File "/home/rgommers/code/pixi-dev-scipystack/scipy/.pixi/envs/free-threading/lib/python3.13t/threading.py", line 316 in _acquire_restore
  File "/home/rgommers/code/pixi-dev-scipystack/scipy/.pixi/envs/free-threading/lib/python3.13t/threading.py", line 368 in wait
  File "/home/rgommers/code/pixi-dev-scipystack/scipy/.pixi/envs/free-threading/lib/python3.13t/threading.py", line 394 in wait_for
  File "/home/rgommers/code/pixi-dev-scipystack/scipy/.pixi/envs/free-threading/lib/python3.13t/threading.py", line 764 in _wait
  File "/home/rgommers/code/pixi-dev-scipystack/scipy/.pixi/envs/free-threading/lib/python3.13t/threading.py", line 729 in wait
  File "/home/rgommers/code/tmp/pytest-run-parallel/src/pytest_run_parallel/plugin.py", line 57 in closure
  File "/home/rgommers/code/pixi-dev-scipystack/scipy/.pixi/envs/free-threading/lib/python3.13t/threading.py", line 992 in run
  File "/home/rgommers/code/pixi-dev-scipystack/scipy/.pixi/envs/free-threading/lib/python3.13t/threading.py", line 1041 in _bootstrap_inner
  File "/home/rgommers/code/pixi-dev-scipystack/scipy/.pixi/envs/free-threading/lib/python3.13t/threading.py", line 1012 in _bootstrap

Thread 0x00007058da7fc6c0 (most recent call first):
  File "/home/rgommers/code/pixi-dev-scipystack/scipy/.pixi/envs/free-threading/lib/python3.13t/site-packages/numpy/_core/fromnumeric.py", line 52 in _wrapfunc
  File ???Fatal Python error: Segmentation fault
```

A lock in the tests isn't a fix though. This looks like a thread safety issue in Pythran for strided input. What should be done with this is turn the test into a self-contained reproducer and report it as a Pythran issue.